Source/Core/InputCommon/ControllerInterface/Device.cpp

```diff
@@ -26,45 +27,38 @@ namespace Core
 //
 Device::~Device()
 {
-  // delete inputs
-  for (Device::Input* input : m_inputs)
-    delete input;
-
-  // delete outputs
-  for (Device::Output* output : m_outputs)
-    delete output;
 }
 
-void Device::AddInput(Device::Input* const i)
+void Device::AddInput(std::unique_ptr<Input> input)
 {
-  m_inputs.push_back(i);
+  m_inputs.push_back(std::move(input));
 }
 
-void Device::AddOutput(Device::Output* const o)
+void Device::AddOutput(std::unique_ptr<Output> output)
 {
-  m_outputs.push_back(o);
+  m_outputs.push_back(std::move(output));
 }
 
 Device::Input* Device::FindInput(const std::string& name) const
```

Comment:
Are these returned raw pointers ever used in a different thread from the one that owns the Device object? If so, maybe this should all be made shared_ptrs instead of unique_ptrs. Your thoughts?